TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/util/ActiveContextMonitor.java

```diff
@@ -0,0 +1,263 @@
+                                Logger logger) {
+        this(globalContextTracker, perActivityContextTracker, orderedRequestTracker, totalItemsToOutputLimit,
+                formatWorkItem, (level, supplier)->logger.atLevel(level).setMessage(supplier).log(),
+                logger::isEnabledForLevel);
+    }
+
+    public ActiveContextMonitor(ActiveContextTracker globalContextTracker,
+                                ActiveContextTrackerByActivityType perActivityContextTracker,
+                                OrderedWorkerTracker<Void> orderedRequestTracker,
+                                int totalItemsToOutputLimit,
+                                Function<DiagnosticTrackableCompletableFuture<String, Void>, String> formatWorkItem,
+                                BiConsumer<Level, Supplier<String>> logger,
+                                Predicate<Level> logLevelIsEnabled) {
+        this(globalContextTracker, perActivityContextTracker, orderedRequestTracker, totalItemsToOutputLimit,
+                formatWorkItem, logger, logLevelIsEnabled, Map.of(
+                        Level.ERROR, Duration.ofSeconds(600),
+                        Level.WARN, Duration.ofSeconds(60),
+                        Level.INFO, Duration.ofSeconds(30),
+                        Level.DEBUG, Duration.ofSeconds(5),
+                        Level.TRACE, Duration.ofSeconds(2)));
+    }
+
+    public ActiveContextMonitor(ActiveContextTracker globalContextTracker,
+                                ActiveContextTrackerByActivityType perActivityContextTracker,
+                                OrderedWorkerTracker<Void> orderedRequestTracker,
+                                int totalItemsToOutputLimit,
+                                Function<DiagnosticTrackableCompletableFuture<String, Void>, String> formatWorkItem,
+                                BiConsumer<Level, Supplier<String>> logger,
+                                Predicate<Level> logLevelIsEnabled,
+                                Map<Level,Duration> levelShowsAgeOlderThanMap) {
+
+        this.globalContextTracker = globalContextTracker;
+        this.perActivityContextTracker = perActivityContextTracker;
+        this.orderedRequestTracker = orderedRequestTracker;
+        this.totalItemsToOutputLimit = totalItemsToOutputLimit;
+        this.logger = logger;
+        this.formatWorkItem = formatWorkItem;
+        this.logLevelIsEnabled = logLevelIsEnabled;
+        ageToLevelEdgeMapRef = new AtomicReference<>();
+        setAgeToLevelMap(levelShowsAgeOlderThanMap);
+    }
+
+    public void setAgeToLevelMap(Map<Level, Duration> levelShowsAgeOlderThanMap) {
+        ageToLevelEdgeMapRef.set(levelShowsAgeOlderThanMap.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getValue, Map.Entry::getKey,
+                        (x,y) -> { throw Lombok.sneakyThrow(new IllegalStateException("Shouldn't have any merges")); },
+                        TreeMap::new)));
+    }
```

Comment:
Can you explain what's going on here?

Reply:
I needed the merge function so that I could specify what kind of map to create.
I've changed the code to just make two maps
```
        ageToLevelEdgeMapRef.set(new TreeMap<>(levelShowsAgeOlderThanMap.entrySet().stream()
                .collect(Collectors.toMap(Map.Entry::getValue, Map.Entry::getKey))));
```

It is called in the constructor and within unit tests and the number of items is 5.  Hopefully that's easier to read & worth the 'cost'.